source/CMakeLists.txt

```diff
@@ -72,6 +72,9 @@ endif( OpENer_TESTS )
 # OpENer C flags                      #
 #######################################
 set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall" )
+# The following extra flags are needed to compile without warnings on mingw
+# TODO: Fix the root problems and remove this line
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-incompatible-pointer-types -Wno-pointer-sign -Wno-format -Wno-discarded-qualifiers -Wno-unknown-pragmas" )
```

Comment:
Please do not disable warning, especially not for all platforms. Also, please do not put TODOs into source files, I currently try to get rid of them. Either this is an issue or it is not